File: 백준/Bronze/3052. 나머지/나머지.cc
#include <iostream>
#include <vector>

int main() {
    std::vector<bool> arr(42, false);
    int num, result = 0;

    for (int i = 0; i < 10; i++) {
        std::cin >> num;
        int temp = num % 42;

        if (!arr[temp]) {
            arr[temp] = true;
            result++;
        }
    }

    std::cout << result << std::endl;
    return 0;
}

File: 백준/Bronze/3052. 나머지/나머지.cs
using System;

class Program {
    static void Main() {
        bool[] arr = new bool[42];
        int result = 0;

        for (int i = 0; i < 10; i++) {
            int num = int.Parse(Console.ReadLine());
            int temp = num % 42;

            if (!arr[temp]) {
                arr[temp] = true;
                result++;
            }
        }

        Console.WriteLine(result);
    }
}